pkg/restore-util/range.go

```diff
@@ -33,6 +36,47 @@ func (r *Range) contains(key []byte) bool {
 		(len(end) == 0 || bytes.Compare(key, end) < 0)
 }
 
+// sortRanges checks if the range overlapped and sort them
+func sortRanges(ranges []Range, rewriteRules *RewriteRules) ([]Range, error) {
+	rangeTree := NewRangeTree()
+	for _, rg := range ranges {
+		if rewriteRules != nil {
+			var startRule *import_sstpb.RewriteRule
+			rg.StartKey, startRule = replacePrefix(rg.StartKey, rewriteRules)
+			if startRule == nil {
+				log.Warn("cannot find rewrite rule", zap.ByteString("key", rg.StartKey))
+			} else {
+				log.Debug(
+					"rewrite start key",
+					zap.ByteString("key", rg.StartKey),
+					zap.Stringer("rule", startRule))
+			}
+			var endRule *import_sstpb.RewriteRule
+			rg.EndKey, endRule = replacePrefix(rg.EndKey, rewriteRules)
+			if endRule == nil {
+				log.Warn("cannot find rewrite rule", zap.ByteString("key", rg.EndKey))
+			} else {
+				log.Debug(
+					"rewrite end key",
+					zap.ByteString("key", rg.EndKey),
+					zap.Stringer("rule", endRule))
+			}
+		}
+		if out := rangeTree.InsertRange(rg); out != nil {
+			return nil, errors.Errorf("ranges overlapped: %v, %v", out.(*Range).String(), rg.String())
```

Comment:
```suggestion
			return nil, errors.Errorf("ranges overlapped: %s, %s", out, rg)
```

`%s` would implicitly call `.String()` for any `Stringer`.